src/client/nav/response.js

```diff
@@ -582,9 +582,17 @@ spf.nav.response.extract_ = function(frag) {
           var url = attr.match(spf.nav.response.AttributeRegEx.SRC);
           url = url ? url[1] : '';
           var async = spf.nav.response.AttributeRegEx.ASYNC.test(attr);
-          result['scripts'].push(
-              {url: url, text: text, name: name, async: async});
-          return '';
+          var type = spf.nav.response.AttributeRegEx.TYPE.exec(attr);
+          var inject = !type || type[1].indexOf('/javascript') > 0 ||
```

Comment:
Please use the `spf.string.contains` function so the compiler can re-use the function call:

```js
var inject = !type || spf.string.contains(type[1], '/javascript') ||
    spf.string.contains(type[1], '/x-javascript') ||
    spf.string.contains(type[1], '/ecmascript');
```